### PR TITLE
EKA-G-HIU-25, EKA-G-HIU-30 | HIU making /consent-requests/init call to Gateway, Handling callback /on-init from Gateway

### PR DESCRIPTION
**Pull request by angshu**

### CODE REVIEW

File: src/main/java/in/org/projecteka/hiu/SecurityConfiguration.java [hunk 1]
```diff
@@ -52,6 +52,7 @@ public SecurityWebFilterChain securityWebFilterChain(
                                            "/**.css",
                                            "/**.png",
                                            "/health-information/fetch/**/attachments/**",
+                                           "/v1/consent-requests/on-init",
```
Comment (dheeraj-p) [inline src/main/java/in/org/projecteka/hiu/SecurityConfiguration.java]: It shouldn't be in `WHITELISTED_URLS` as we have authentication on this. It should rather be in `SERVICE_ONLY_URLS`, if I am not wrong.

Comment (dheeraj-p) [reply src/main/java/in/org/projecteka/hiu/SecurityConfiguration.java]: Please let me know if I am missing any context.

File: src/main/java/in/org/projecteka/hiu/consent/ConsentRepository.java [hunk 2]
```diff
@@ -36,21 +36,32 @@
             " VALUES ($1, $2, $3, $4, $5)";
     private static final String UPDATE_CONSENT_ARTEFACT_STATUS_QUERY = "UPDATE " +
             "consent_artefact set status=$1, date_modified=$2 where consent_artefact_id=$3";
+    //TODO
     private static final String INSERT_CONSENT_REQUEST_QUERY = "INSERT INTO " +
             "consent_request (consent_request, consent_request_id) VALUES ($1, $2)";
+    //TODO
```
Comment (dheeraj-p) [inline src/main/java/in/org/projecteka/hiu/consent/ConsentRepository.java]: TODO?

File: src/main/java/in/org/projecteka/hiu/consent/ConsentService.java [hunk 3]
```diff
@@ -205,4 +229,68 @@
...
+                    response.getResp().getRequestId(),
+                    ConsentStatus.ERRORED,
+                    "");
+        }
+
+        if (response.getConsentRequest() != null) {
+            return consentRepository.consentRequestStatus(response.getResp().getRequestId())
+                    .switchIfEmpty(Mono.error(consentRequestNotFound()))
+                    .flatMap(status -> updateConsentRequestStatus(response, status));
+        } else {
```
Comment (dheeraj-p) [inline src/main/java/in/org/projecteka/hiu/consent/ConsentService.java]: We don't actually need this `else` block here as we are returning from `if` block.

Comment (angshu) [reply src/main/java/in/org/projecteka/hiu/consent/ConsentService.java]: done

File: src/main/java/in/org/projecteka/hiu/consent/ConsentRepository.java (hunk 2 above)
Comment (angshu) [reply src/main/java/in/org/projecteka/hiu/consent/ConsentRepository.java]: left comments. mostly for future refactoring when we work on the /consents/notify

**Review by dheeraj-p — APPROVED** [review]

**Pull request merged**